src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/AssemblyVersionValueProvider.cs

```diff
@@ -22,7 +22,7 @@ internal sealed class AssemblyVersionValueProvider : BaseVersionValueProvider
                 return s_DefaultAssemblyVersion;
             }
 
-            return new Version(defaultVersion.Major, defaultVersion.Minor, defaultVersion.Build, revision: 0);
+            return new Version(defaultVersion.Major, defaultVersion.Minor, Math.Max(defaultVersion.Build, 0), revision: Math.Max(defaultVersion.Revision, 0));
```

Comment:
Yes, it tries to replicate what the build will emit. Hope we can get rid of this soon. If this bug is not accepted for RTM, I'd prefer us fixing the SDK bug and getting rid of the interceptors, instead of taking this PR.